test/unit/cuda/test-atomic.cpp

```diff
@@ -0,0 +1,257 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
```

Comment:
I'm fine with that. I could arrange them all in one unit test file and get more reuse.